packages/block-editor/src/components/rich-text/index.js

```diff
@@ -378,7 +368,20 @@ export class RichText extends Component {
 		}
 
 		this.recalculateBoundaryStyle();
-		this.onSelectionChange();
+
+		// We know for certain that on focus, the old selection is invalid. It
```

Comment:
Perhaps it's not worth worrying about, but: There are cases where a `focus` event occurs, but the selection remains the same. Notably, when the tab loses or regains focus (becomes the active tab again). I notice it most when debugging, to add a breakpoint in a code path which triggers on a change to state. The act of focusing DevTools and changing back to the window is enough to cause a change in state, so the breakpoint will fire.

Perhaps it's not a big deal, but could we:

- Avoid calling `onSelectionChange` if the selection indices are the same? Perhaps also avoid `setState` as well.
- Update the reducer for `selectionStart` and `selectionEnd` to avoid [creating new object references](https://github.com/WordPress/gutenberg/blob/7a337cfeaaade6622d6bc9a08101b1b79a1579ea/packages/block-editor/src/store/reducer.js#L1021-L1025) if the updated offsets are the same as the current state value?

Reply:
@aduth `onSelectionChange` already avoids setting state if the selection is the same. I guess the problem here is that it's set to `undefined`?

Reply:
> @aduth `onSelectionChange` already avoids setting state if the selection is the same

Where does this happen?

Reply:
Here: 

https://github.com/WordPress/gutenberg/blob/5957797727dd324085270bcc56c1168c381e212d/packages/rich-text/src/component/index.js#L537-L545